test/test-eintr-handling.c

```diff
@@ -0,0 +1,63 @@
+/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+#include <unistd.h>
+
+static uv_loop_t* loop;
+static uv_fs_t read_req;
+
+static char buf[32];
+static uv_buf_t iov;
+
+struct thread_ctx {
+  uv_barrier_t barrier;
+  int fd;
+};
+
+static void thread_main(void* arg) {
+  kill(getpid(), SIGUSR1);
+}
+
+void sig_func(int sig) {
+  printf("Caught signal: %d\n",sig);
+}
+
+TEST_IMPL(eintr_handling) {
+  struct thread_ctx ctx;
+  uv_thread_t thread;
+
+  struct sigaction int_handler = {.sa_handler=sig_func};
+  sigaction(SIGUSR1,&int_handler, 0);
+
+  iov = uv_buf_init(buf, sizeof(buf));
+  loop = uv_default_loop();
+
+  printf("spawning kill thread\n");
+
+  ASSERT(0 == uv_thread_create(&thread, thread_main, &ctx));
+
+  uv_fs_read(loop, &read_req, 0, &iov, 1, -1, NULL);
```

Comment:
I think you may want to create a `pipe` by calling `pipe()`. Read on one side of it, and write on another side when sending a singal.